### PR TITLE
Adding git stash drop for issue 76195

### PR DESCRIPTION
**Pull request by drew212**

### CODE REVIEW

**Comment by drew212** [comment]
This is a fix for issue #76195

**Comment by drew212** [comment]
@joaomoreno should the menu item have "..." since this is an action that results in the drop-down menu to select the stash to drop, not sure about the style here.

**Comment by joaomoreno** [comment]
> @joaomoreno should the menu item have "..." since this is an action that results in the drop-down menu to select the stash to drop, not sure about the style here.

Yeah, that's the usual pattern.

**Comment by drew212** [comment]
Ok, done. Let me know if there are other changes you recommend.

File: extensions/git/package.json [hunk 1]
```diff
@@ -642,6 +647,10 @@
         {
           "command": "git.stashApplyLatest",
           "when": "config.git.enabled && gitOpenRepositoryCount != 0"
+        },
+        {
+          "command": "git.dropStash",
+          "when": "config.get.enabled && gitOpeneRepositoryCount != 0"
```
Comment (ZuBB) [inline extensions/git/package.json]: > `gitOpeneRepositoryCount`

> `opene`

> `e`

Seems like a typo

Comment (drew212) [reply extensions/git/package.json]: Yes, it is, thank you!

File: extensions/git/package.json [hunk 2]
```diff
@@ -391,6 +391,11 @@
         "command": "git.stashApplyLatest",
         "title": "%command.stashApplyLatest%",
         "category": "Git"
+      },
+      {
+        "command": "git.dropStash",
```
Comment (Thiht) [inline extensions/git/package.json]: shouldn't this be `stashDrop` to be consistent with the other commands? (eg. `git.stashPop`, `git.stashApply`, ...)

Comment (drew212) [reply extensions/git/package.json]: Good point, I've updated the naming scheme 👍 .

File: extensions/git/src/git.ts [hunk 3]
```diff
@@ -1526,6 +1526,24 @@ export class Repository {
 		}
 	}
 
+	async dropStash(index?: number): Promise<void> {
```
Comment (joaomoreno) [inline extensions/git/src/git.ts]: If there's no need, let's not make this argument optional.

File: extensions/git/package.json (hunk 2 above)
Comment (joaomoreno) [reply extensions/git/package.json]: Did you forget to push this?

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: extensions/git/src/git.ts (hunk 3 above)
Comment (drew212) [reply extensions/git/src/git.ts]: Done.

File: extensions/git/src/git.ts [hunk 4]
```diff
@@ -1526,6 +1526,24 @@ export class Repository {
...
+		const args = ['stash', 'drop'];
+
+		try {
+			if (typeof index === 'number') {
+				args.push(`stash@{${index}}`);
+
+				await this.run(args);
```
[the remark this answers is not in the document]
Comment (drew212) [reply extensions/git/src/git.ts]: Fixed.

**Comment by MJ-Mohith** [comment]
This is really a great feature. I am looking forward to use this feature in our everyday's workflow.

**Comment by drew212** [comment]
> 
> 
> This is really a great feature. I am looking forward to use this feature in our everyday's workflow.
> Will it also enable selective drop of stashed contents?

Yes! The way I've designed this you can selectively drop a stash. Waiting on @joaomoreno to approve or request more changes.

File: extensions/git/src/git.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (joaomoreno) [reply extensions/git/src/git.ts]: This caused syntax errors.

image

File: extensions/git/src/repository.ts [hunk 5]
```diff
@@ -1209,6 +1209,10 @@ export class Repository implements Disposable {
 		return await this.run(Operation.Stash, () => this.repository.popStash(index));
 	}
 
+	async dropStash(index?: number): Promise<void> {
+		return await this.run(Operation.Stash, () => this.repository.dropStash(index));
```
Comment (joaomoreno) [inline extensions/git/src/repository.ts]: Compile error here, since `index` can be `undefined`.

File: extensions/git/package.json [hunk 6]
```diff
@@ -642,6 +647,10 @@
         {
           "command": "git.stashApplyLatest",
           "when": "config.git.enabled && gitOpenRepositoryCount != 0"
+        },
+        {
+          "command": "git.stashDrop",
+          "when": "config.get.enabled && gitOpenRepositoryCount != 0"
```
Comment (joaomoreno) [inline extensions/git/package.json]: Typo here: should be `git`, not `get`.

File: extensions/git/src/git.ts [hunk 7]
```diff
@@ -1526,6 +1526,24 @@ export class Repository {
...
+	async dropStash(index: number): Promise<void> {
+		const args = ['stash', 'drop'];
+
+		if (typeof index === 'number') {
+			args.push(`stash@{${index}}`);
+
+			try {
+				await this.run(args);
+			}
+		} catch (err) {
```
Comment (joaomoreno) [inline extensions/git/src/git.ts]: There's a syntax error here.

image

**Comment by joaomoreno** [comment]
There were more issues which I've commented on. But I went ahead and fixed them and merged the PR. Thanks! :beers:

**Pull request merged**